java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/server/TaskManagerServer.java

```diff
@@ -16,34 +16,66 @@
 
-    @BrpcMeta(serviceName = "openmldb.taskmanager.TaskManagerServer", methodName = "StopJob")
-    TaskManager.StopJobResponse StopJob(TaskManager.StopJobRequest request);
+    public TaskManagerServer() {
+        try {
+            FailoverWatcher failoverWatcher = new FailoverWatcher();
 
-    @BrpcMeta(serviceName = "openmldb.taskmanager.TaskManagerServer", methodName = "DeleteJob")
-    TaskManager.DeleteJobResponse DeleteJob(TaskManager.DeleteJobRequest request);
+            logger.info("The server runs and prepares for leader election");
+            if (failoverWatcher.blockUntilActive()) {
+                logger.info("The server becomes active master and prepare to do business logic");
+                runBrpcServer();
+            }
+            failoverWatcher.close();
+            logger.info("The server exits after running business logic");
 
-    @BrpcMeta(serviceName = "openmldb.taskmanager.TaskManagerServer", methodName = "ShowBatchVersion")
-    TaskManager.ShowJobResponse ShowBatchVersion(TaskManager.ShowBatchVersionRequest request);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
 
-    @BrpcMeta(serviceName = "openmldb.taskmanager.TaskManagerServer", methodName = "RunBatchSql")
-    TaskManager.ShowJobResponse RunBatchSql(TaskManager.RunBatchSqlRequest request);
+    public void runBrpcServer() {
+        try {
+            RpcServerOptions options = new RpcServerOptions();
+            options.setReceiveBufferSize(64 * 1024 * 1024);
+            options.setSendBufferSize(64 * 1024 * 1024);
+            options.setIoThreadNum(TaskManagerConfig.WORKER_THREAD);
+            options.setWorkThreadNum(TaskManagerConfig.IO_THREAD);
+            final RpcServer rpcServer = new RpcServer(TaskManagerConfig.PORT, options);
+            rpcServer.registerService(new TaskManagerImpl());
+            rpcServer.start();
+            log.info("Start TaskManager on {} with worker thread number {}", TaskManagerConfig.PORT, TaskManagerConfig.WORKER_THREAD);
 
-    @BrpcMeta(serviceName = "openmldb.taskmanager.TaskManagerServer", methodName = "RunBatchAndShow")
-    TaskManager.ShowJobResponse RunBatchAndShow(TaskManager.RunBatchAndShowRequest request);
+            // make server keep running
+            synchronized (TaskManagerServer.class) {
+                try {
+                    TaskManagerServer.class.wait();
+                } catch (Throwable e) {
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+            log.error("Fail to start TaskManager, " + e.getMessage());
+        }
+    }
 
-    @BrpcMeta(serviceName = "openmldb.taskmanager.TaskManagerServer", methodName = "ImportOnlineData")
-    TaskManager.ShowJobResponse ImportOnlineData(TaskManager.ImportOnlineDataRequest request);
+    public static void main(String[] args) {
+        TaskManagerServer server = new TaskManagerServer();
+        server.runBrpcServer();
```

Comment:
`runBrpcServer` has been called in `TaskManagerServer` constructor, why call this function again here

Reply:
Right. We should not start the server in constructor and we will rename the method instead of constructor.